performance/shared/src/main/kotlin/org/jetbrains/benchmarksLauncher/launcher.kt

```diff
@@ -60,6 +67,72 @@ abstract class Launcher {
         }
     }
 
+    fun runBenchmarkRepeatedly(logger: Logger,
+                               numWarmIterations: Int,
+                               numberOfAttempts: Int,
+                               name: String,
+                               recordMeasurement: (RecordTimeMeasurement) -> Unit,
+                               benchmarkInstance: Any?,
+                               benchmark: AbstractBenchmarkEntry) {
+        logger.log("Warm up iterations for benchmark $name\n")
+        runBenchmark(benchmarkInstance, benchmark, numWarmIterations)
+        var autoEvaluatedNumberOfMeasureIteration = 1
+        while (true) {
+            var j = autoEvaluatedNumberOfMeasureIteration
+            val time = runBenchmark(benchmarkInstance, benchmark, j)
+            if (time >= 100L * 1_000_000) // 100ms
+                break
+            autoEvaluatedNumberOfMeasureIteration *= 2
+        }
+        logger.log("Running benchmark $name ")
+        for (k in 0..numberOfAttempts) {
+            logger.log(".", usePrefix = false)
+            var i = autoEvaluatedNumberOfMeasureIteration
+            val time = runBenchmark(benchmarkInstance, benchmark, i)
+            val scaledTime = time * 1.0 / autoEvaluatedNumberOfMeasureIteration
+            // Save benchmark object
+            recordMeasurement(RecordTimeMeasurement(BenchmarkResult.Status.PASSED, k, numWarmIterations, scaledTime))
+        }
+        logger.log("\n", usePrefix = false)
+    }
+
+    fun runBenchmark(logger: Logger,
+                     numWarmIterations: Int,
+                     numberOfAttempts: Int,
+                     name: String,
+                     recordMeasurement: (RecordTimeMeasurement) -> Unit,
+                     benchmark: AbstractBenchmarkEntry) {
+        when (benchmark) {
+            is BenchmarkEntryWithInit -> {
+                val benchmarkInstance = benchmark.ctor?.invoke()
+                runBenchmarkRepeatedly(logger,
+                                       numWarmIterations,
+                                       numberOfAttempts,
+                                       name,
+                                       recordMeasurement,
+                                       benchmarkInstance,
+                                       benchmark)
+            }
+            is BenchmarkEntry -> {
+                runBenchmarkRepeatedly(logger,
+                                       numWarmIterations,
+                                       numberOfAttempts,
+                                       name,
+                                       recordMeasurement,
+                                       null,
+                                       benchmark)
+            }
+            is BenchmarkEntryManual -> {
+                logger.log("Running manual benchmark $name")
+                val result = benchmark.lambda()
+                for ((i, durationNs) in result.durationsNs.withIndex()) {
```

Comment:
Are you sure that this new type still is needed? Only to get time/500?

Reply:
I need these benchmarks not to run this code https://github.com/JetBrains/kotlin-native/pull/4018#diff-cb51810e983e4cc6a7fbd6f9e4e653b1R80 On the other hand, it can be solved by introducing a new runtime flag for the launcher to disable automatic iteration count calculation. What do you think?